src/software/new_geom/line.cpp

```diff
@@ -0,0 +1,30 @@
+#include "software/new_geom/line.h"
+
+Line::Line() {}
+
+Line::Line(double y_intercept, double slope) : line(Polynomial({y_intercept, slope})) {}
```

Comment:
It seems like with the interface of `y_intercept, slope` we can't create lines with infinite slope / running along the y-axis? I think this is something we would reasonably want, but it doesn't seem like polynomial can handle that anyway. Maybe we can't even use Polynomial composition here?

Reply:
Seems like the old `Line` class doesn't handle this in a special way either, where if you call `slope` for a vertical `Line` it would just return `inf`. I believe then this would work similarly with this new `Line` class? But conceptually it doesn't make sense for a polynomial line to have an infinite slope, and evaluating it with `valueAt` wouldn't make much sense either. If handling infinite slope is important, we could just use the old representation of `Line` with two `Point`s.

Reply:
I'm inclined to say that if we want a 2D line like this, we should use two lines parameterized as a function of some third variable. ie. `x(t)` and `y(t)`.

Reply:
I think it is important to be able to represent infinite lines at _all_ orientations. It would be totally reasonable to want a line running along the y axis (maybe we're checking something with the goal-line or halfway line).

Most of the useful things we do with lines are with utility functions (that I don't think rely too much on the slope), so maybe we don't actually need access to the slope? I checked and the only place it is used is in unit tests for the ball filter, which can probably be changed to use something else.

Ideas / thoughts on making the lines able to be vertical? Maybe use a more mathematical ilne representation? Sorry for dragging this out just want to make sure we end up with the implementation that suits our needs the best.

Reply:
All good. If we want lines orientable in all directions, `y = mx + b` is not an appropriate representation. We should instead consider a line of the form indicated above, ie `f(t) = <x(t),y(t)>`

Reply:
I'm not even sure "Slope intercept" line would be useful because of how we generally use this geometry on the field, the y-intercept isn't that useful, and slope limits our orientations (in a way). Sorry I didn't think of this earlier.

Reply:
I'm fine with just having a `Line` class that is defined by the `f(t)` above. Sorry for all the hassle with this @jsn0716 - does this make sense?

Reply:
Yep makes sense. So after this change, if a user wants a line with slope + y-intercept then they'll just create a first order `Polynomial`.

Reply:
Yes, I think that's reasonable.

Reply:
Finally got around to updating this, sorry for the delay! The implementation has been changed to a parametric representation. Since we are wanting `Line` to be able to represent a vertical line, it might make sense to change `getSlope` to `getDirection` or `getAngle` instead. Let me know what you think